static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -0,0 +1,90 @@
+define([
+    'common/utils/fastdom-promise',
+    'Promise',
+    'common/utils/$',
+    'common/utils/mediator',
+    'common/utils/detect'
+], function (
+    fastdom,
+    Promise,
+    $,
+    mediator,
+    detect) {
+
+    return function () {
+        if (detect.getBreakpoint() !== 'mobile') {
+            fastdom.read(function () {
+                var adId = 'dfp-ad--top-above-nav',
+                    $adBanner = $('.js-top-banner-above-nav'),
+                    $header = $('.js-header'),
+                    oldAdHeight = $adBanner.height(),
+                    headerHeight = $header.height();
+
+                var topAdRenderedPromise = new Promise(function (resolve) {
+                    mediator.on('modules:commercial:dfp:rendered', function (event) {
+                        if (event.slot.getSlotElementId() === adId) {
+                            resolve();
+                        }
+                    });
+                });
+
+                var oldAdHeightPromise = Promise.resolve(oldAdHeight);
+                var newAdHeightPromise = topAdRenderedPromise.then(function () {
+                    return fastdom.read(function () {
+                        var adSlotPadding = parseInt($('#' + adId, $adBanner).css('padding-bottom')) * 2;
+                        // We must read the iframe attribute height to avoid reading the clientHeight mid-transition
+                        return Number($('iframe', $adBanner).attr('height')) + adSlotPadding;
+                    });
+                });
+
+                var getLatestAdHeight = function () {
+                    return Promise.race([newAdHeightPromise, oldAdHeightPromise]);
+                };
+
+                var render = function () {
+                    getLatestAdHeight().then(function (adHeight) {
+                        fastdom.write(function () {
+                            if (window.scrollY === 0) {
+                                // Reset
+                                $header.css('margin-top', '');
```

Comment:
for readability I'd consider splitting the css changes into functions with a useful name